core/src/main/java/org/ehcache/core/spi/service/LocalPersistenceService.java

```diff
@@ -17,22 +17,71 @@
 package org.ehcache.core.spi.service;
 
 import org.ehcache.CachePersistenceException;
-import org.ehcache.spi.persistence.PersistableResourceService;
+import org.ehcache.spi.service.MaintainableService;
 
 /**
- * Service to provide persistence context to caches requiring it.
- * <P>
- * Will be used by caches with a disk store, whether or not the data should survive a program restart.
- * <P/>
+ * Service that provides unique local persistence spaces (and safe sub spaces) to any service that requires it
+ * under the local root directory.
+ *
+ * @author RKAV refactored as part of issue-1192
  */
-public interface LocalPersistenceService extends PersistableResourceService {
+public interface LocalPersistenceService extends MaintainableService {
+
+  /**
+   * Prepare a logical safe directory space for the owner and return an identifying space Id.
+   *
+   * @param owner Service owner that owns the safe space.
+   * @param name Identifying name for the space.
+   *
+   * @return Identifier that can be used to identify the safe space.
+   */
+  String prepareSafeSpace(String owner, String name);
```

Comment:
In case there are both frs and disk (and maybe hybrid) under the same cache manager. So when I destroy one it does not destroy the other..